source/common/http/http2/codec_impl.cc

```diff
@@ -640,7 +641,8 @@ void ConnectionImpl::StreamImpl::resetStream(StreamResetReason reason) {
   // If we submit a reset, nghttp2 will cancel outbound frames that have not yet been sent.
   // We want these frames to go out so we defer the reset until we send all of the frames that
   // end the local stream.
-  if (useDeferredReset() && local_end_stream_ && !local_end_stream_sent_) {
+  if (useDeferredReset() && local_end_stream_ && !local_end_stream_sent_ &&
+      reason != StreamResetReason::OverloadManager) {
```

Comment:
Can you add more comments here about why we don't do this if the reason is via the overload manager?

Reply:
The thinking is if we're trying to shed memory ASAP if we're resetting the stream due to Overload Manager.

Reply:
OK makes sense. I would just beef up the comment.

Reply:
Done.